bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/IDFDebugLaunchDescriptorType.java

```diff
@@ -66,21 +67,21 @@ public ILaunchDescriptor getDescriptor(Object launchObject)
 		return null;
 	}
 
-	private IProject getProject()
+	private Optional<IProject> getMappedProject(ILaunchConfiguration config)
```

Comment:
this method can be extracted to the separate class and be reused in other places like Launch Config descriptor